integration/run.js

```diff
@@ -20,20 +21,45 @@ const common = {
     common.summaryMap( res )
           .forEach( console.dir );
   },
-  simpleTokens: ( tokens, includePosition ) => {
-    return tokens.map( t => {
-      return (!!includePosition ? t.position + ':' : '') + t.token;
+  bucketTokens: tokens => {
+    const positions = {};
+    tokens.forEach((t, i) => {
+      // format returned by elasticsearch
+      if (_.isPlainObject(t)) {
+        const pos = '@pos' + t.position;
+        if (!positions[pos]) { positions[pos] = []; }
+        positions[pos].push(t.token);
+      }
+      // 'simple tokens' format
+      // eg '1:foo'
+      else if (_.isString(t)) {
+        const match = t.match(/^(\d+):(.+)$/);
+        let pos = '@pos' + i;
+        if (match) {
+          pos = '@pos' + match[1];
+          t = match[2];
+        }
+        if (!positions[pos]) { positions[pos] = []; }
+        positions[pos].push(t);
+      }
     });
+    // sort all the arrays so that order is irrelevant
+    for (var attr in positions){
+      positions[attr] = positions[attr].sort();
+    }
+    return positions;
   },
-  analyze: ( suite, t, analyzer, comment, text, expected, includePosition ) => {
-    suite.assert( done => {
+  analyze: (suite, t, analyzer, comment, text, expected) => {
+    suite.assert(done => {
       suite.client.indices.analyze({
         index: suite.props.index,
-        analyzer: analyzer,
-        text: text
-      }, ( err, res ) => {
-        if( err ){ console.error( err ); }
-        t.deepEqual( common.simpleTokens( res.tokens, includePosition ), expected, comment );
+        body: {
+          analyzer: analyzer,
+          text: text.toString()
+        }
```

Comment:
this change was also interesting and is required in ES6.

I believe prior to this change the request was being made via query-string and now it's being made in the request body.

The interesting bit is when I change this some of the tests which handle commas started failing, so I think there is either a bug in the elasticsearch-js handling of this or something else weird was going on.

Regardless, this is the better form to use.

Reply:
Check this out, the second token position changes depending on the request method:

```js
GET http://localhost:9200/pelias/_analyze?analyzer=peliasIndexOneEdgeGram&text=a%2Cb

{
    "tokens": [
        {
            "token": "a",
            "start_offset": 0,
            "end_offset": 1,
            "type": "word",
            "position": 0
        },
        {
            "token": "b",
            "start_offset": 2,
            "end_offset": 3,
            "type": "word",
            "position": 101
        }
    ]
}
```

```js
GET http://localhost:9200/pelias/_analyze
{
	"analyzer": "peliasIndexOneEdgeGram",
	"text": "a,b"
}

{
    "tokens": [
        {
            "token": "a",
            "start_offset": 0,
            "end_offset": 1,
            "type": "word",
            "position": 0
        },
        {
            "token": "b",
            "start_offset": 2,
            "end_offset": 3,
            "type": "word",
            "position": 1
        }
    ]
}
```

Reply:
They removed the query-string version in ES6, so it's probably a known bug.

Reply:
Pretty wild. So it has different behavior in ES6, but I assume the two queries would do the same thing in ES5? Or is there something else going on here?